ruby/rspec/style_guide.md

```diff
@@ -0,0 +1,645 @@
+      it 'returns the summary' do
+        # ...
+      end
+    end
+  end
+end
+```
+
+### `let`, `subject`, and `before`/`after` group line returns
+
+Leave one line return after `let`, `subject`, and `before`/`after` blocks.
+
+#### Bad Example
+
+```ruby
+describe Article do
+  subject { create(:some_article) }
+  describe '#summary' do
+    # ...
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+describe Article do
+  subject { create(:some_article) }
+
+  describe '#summary' do
+    # ...
+  end
+end
+```
+
+### `let`, `subject`, `before`/`after` grouping
+
+Only group `let`, `subject` blocks and separate them from `before`/`after`
+blocks. It makes the code much more readable.
+
+#### Bad Example
+
+```ruby
+describe Article do
+  subject { create(:some_article) }
+  let(:user) { create(:user) }
+  before do
+    # ...
+  end
+  after do
+    # ...
+  end
+  describe '#summary' do
+    # ...
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+describe Article do
+  subject { create(:some_article) }
+  let(:user) { create(:user) }
+
+  before do
+    # ...
+  end
+
+  after do
+    # ...
+  end
+
+  describe '#summary' do
+    # ...
+  end
+end
+```
+
+### `it` block line returns
+
+Leave one line return around `it` blocks. This helps to separate the
+expectations from their conditional logic (contexts for instance).
+
+#### Bad Example
+
+```ruby
+describe '#summary' do
+  let(:item) { double('something') }
+
+  it 'returns the summary' do
+    # ...
+  end
+  it 'does something else' do
+    # ...
+  end
+  it 'does another thing' do
+    # ...
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+describe '#summary' do
+  let(:item) { double('something') }
+
+  it 'returns the summary' do
+    # ...
+  end
+
+  it 'does something else' do
+    # ...
+  end
+
+  it 'does another thing' do
+    # ...
+  end
+end
+```
+
+### `before(:each)`or `before`
+
+There is no need to specify `(:each)` for `before`/`after` blocks, as it is
+the default functionality. There are almost zero cases to use `before(:all)`
+anymore, but if you do find one - just write it out like `before(:all)`
+
+#### Bad Example
+
+```ruby
+describe '#summary' do
+  before(:each) do
+    subject.summary = 'something'
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+describe '#summary' do
+  before do
+    subject.summary = 'something'
+  end
+end
+```
+
+### 'should' it or 'should not' in `it` statements
+
+Do not write 'should' or 'should not' in the beginning of your `it` blocks.
+The descriptions represent actual functionality - not what might be happening.
+
+#### Bad Example
+
+```ruby
+it 'should return the summary' do
+  # ...
+end
+```
+
+#### Good Example
+
+```ruby
+it 'returns the summary' do
+  # ...
+end
+```
+
+### The One Expectation
+
+Use only one expectation per example. There are very few scenarios where two
+or more expectations in a single `it` block should be used. So, general rule
+of thumb is one expectation per `it` block.
```

Comment:
What is "thumb" in this context? Just want to make sure it is not a typo.

Reply:
It isn't, it's like saying "the golden rule"